src/main/java/com/google/research/bleth/simulator/StatisticsState.java

```diff
@@ -0,0 +1,109 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+/** A mediator between the statistics the simulation gathered and their storage on datastore. */
+public class StatisticsState {
+    private final String simulationId;
+    private final  Map<String, Double> distanceStats;
+    private final Map<String, Double> beaconsObservedPercent;
+
+    /**
+     * Create a new StatisticsState.
+     * @param simulationId is the simulation id associated with the statistical data.
+     * @param distanceStats is statistics about the difference between the beacons' real locations and their estimated locations.
+     * @param beaconsObservedPercent maps each beacon to the percentage of rounds in which it has been observed.
+     * @return a new instance of StatisticsState
+     */
+    public static StatisticsState create(String simulationId, Map<String, Double> distanceStats, Map<String, Double> beaconsObservedPercent) {
+        checkNotNull(distanceStats);
+        checkNotNull(beaconsObservedPercent);
+        return new StatisticsState(simulationId, distanceStats, beaconsObservedPercent);
+    }
+
+    /**
+     * Create and write multiple datastore entities, each represents an aggregate function such as min and max,
+     * of the distance between the beacons' real locations and their estimated locations.
+     */
+    public void writeDistancesStats() {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        for (Map.Entry<String, Double> entry : distanceStats.entrySet()) {
+            Entity entity = new Entity(Schema.StatisticsState.entityKindDistance);
+            entity.setProperty(Schema.StatisticsState.simulationId, simulationId);
+            entity.setProperty(Schema.StatisticsState.aggregateFunction, entry.getKey());
+            entity.setProperty(Schema.StatisticsState.value, entry.getValue());
+            datastore.put(entity);
+        }
+    }
+
+    /**
+     * Create and write multiple datastore entities, each represents the percentage of rounds a beacon
+     * has been observed by at least one observer, i.e. has been detected by the resolver.
+     */
+    public void writeBeaconsObservedPercentStats() {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        for (String beaconId : beaconsObservedPercent.keySet()) {
+            Entity entity = new Entity(Schema.StatisticsState.entityKindBeaconsObservedPercent);
+            entity.setProperty(Schema.StatisticsState.simulationId, simulationId);
+            entity.setProperty(Schema.StatisticsState.beaconId, beaconId);
+            entity.setProperty(Schema.StatisticsState.percent, beaconsObservedPercent.get(beaconId));
+            datastore.put(entity);
```

Comment:
Do we really want to store this measure for each beacon? 
Even if we do, I would also store the average over all beacons (it is easier to calculate it here rather than by querying the db).